### PR TITLE
Various improvements to multi-Z travel [PORT]

### PR DESCRIPTION
**Pull request by DoctorSquishy**
# About The Pull Request

 "This PR fixes a lot of things about multi-z travel. (Port of https://github.com/BeeStation/NSV13/pull/1418 and https://github.com/BeeStation/NSV13/pull/1894)
It contains several improvements to using stairs, especially with pulling, buckling, and vehicles. Buckled things will stay buckled while transiting stairs, vehicles can go down stairs, and being in a wheelchair no longer makes you immune to fall damage.

It changes travel_z so that moving upwards into space or openspace is okay even if you start on a floor, and moving downwards onto a turf is okay as long as you're starting on space or openspace. It also allows the Move-Upwards and Move-Down verbs to be used for travel_z. It adds handing for some incorporeal beings - the blob overmind (prior to placement), and most things that use jaunt movement, including revenants and the Eminence. Ninja jaunting (unused?) was deliberately excluded because gravity should still apply to ninjas most of the time."

It adds power cables that go up and down in place of multi-Z power relays. https://github.com/discordia-space/CEV-Eris/pull/2032 was used as a reference.

## Why It's Good For The Game

"It makes multi-z much more functional and allows more intuitive multi-z movement."

## TO-DO
- [x] Update the MultiZ wiring for HoleStation
- [x] ~Update the MultiZ wiring for CorgStation~ Corg needs an overhaul.

## Changelog

:cl: DoctorSquishy, covertcorvid, Kmc2000
add: Added stairs to multi-z debug station
add: Power cables that go up and down. Use 2 pieces of cable on an open space to create one.
remove: multi-deck power adapters
fix: Buckles are no longer cancelled when using stairs
fix: Vehicles are now able to go down stairs
fix: Pulling someone in a roller bed or wheelchair on stairs now acts the same regardless of whether you grabbed the person or the object
fix: Fireman carrying people up and down stairs works now
fix: Being buckled to a wheelchair or other vehicle no longer makes you immune to damage from falling through open space
fix: Non-observer ghosts (revenants, eminence, poly's ghost) no longer need to hitch a ride in an elevator to get off their starting level.
tweak: Blob overmind can use Move-Upwards and Move-Down verbs to find a location to place itself
tweak: Upwards movement from a floor tile in zero gravity will work as long as the tile above you is open or space
tweak: The Move-Upwards and Move-Down verbs from the IC tab can be used by more than just observers
tweak: Revenants and Eminences can use the up-down radial menu
tweak: A shuttle that can choose its landing site can land on any level of a multi-Z station
code: Stair travel uses Move or doMove now to better handle pulling and buckling
code: Allow_z_travel now indicates whether a turf is a valid start location for downwards travel, or a valid destination for upwards travel
/:cl:

### CODE REVIEW

**Comment by williamcll** [comment]
You can skip CorgStation, it needs a whole revamp, including swapping the entire Z-level.

**Comment by DoctorSquishy** [comment]
> You can skip CorgStation, it needs a whole revamp, including swapping the entire Z-level.

How's that single Z Corg coming? Probably should just do that one.

**Comment by williamcll** [comment]
> > You can skip CorgStation, it needs a whole revamp, including swapping the entire Z-level.
> 
> How's that single Z Corg coming? Probably should just do that one.

I asked on discord and people wanted Multi-Z.

File: code/game/turfs/open/openspace.dm [hunk 1]
```diff
@@ -54,6 +54,9 @@ GLOBAL_DATUM_INIT(openspace_backdrop_one_for_all, /atom/movable/openspace_backdr
 /turf/open/openspace/can_have_cabling()
 	if(locate(/obj/structure/lattice/catwalk, src))
 		return TRUE
+	var/turf/B = below()
```
Comment (nednaZ) [inline code/game/turfs/open/openspace.dm:57]: Just needs a better name

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -224,7 +230,13 @@ GLOBAL_LIST_EMPTY(station_turfs)
 	if(!do_after(user, 30, FALSE, get_turf(user)))
 		animate(user, 0, flags = ANIMATION_END_NOW)
 		return
-	if(!istype(target, /turf/open/space) && !istype(target, /turf/open/openspace))
+	if(isliving(user))
+		var/mob/living/L = user
```
Comment (nednaZ) [inline code/game/turfs/turf.dm:234]: Variable name

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -296,25 +308,23 @@ GLOBAL_LIST_EMPTY(station_turfs)
 /turf/proc/can_zFall(atom/movable/A, levels = 1, turf/target)
 	return zPassOut(A, DOWN, target) && target.zPassIn(A, DOWN, src)
 
-/turf/proc/zFall(atom/movable/A, levels = 1, force = FALSE)
+/turf/proc/zFall(atom/movable/A, levels = 1, force = FALSE, old_loc = null)
```
Comment (nednaZ) [inline code/game/turfs/turf.dm:311]: Variable name

**Review by nednaZ — COMMENTED** [review]
All I'm seeing are non-descriptive variable names as an issue.
Worth fixing for future coders that go through this

**Comment by nednaZ** [comment]
No issues have been found, looks good.

**Pull request merged**